shtdlib.sh

```diff
@@ -661,6 +661,54 @@ run_dir="${run_dir:-$(dirname "${script_full_path}")}"
 # Default is to clean up after ourselves
 cleanup="${cleanup:-true}"
 
+# Create NSS Wrapper passwd and group files
+# Accepts 3 optional arguments, username, group and home directory
+# Defaults to bob, builders and a temporary directory
+# Note that if a home directory is specified and it's temporary it will need to
+# be removed/cleaned up by the code calling this function
+function init_nss_wrapper {
+    umask_decorator_mask=${NSS_WRAPPED_FILE_MASK:-0002}
+    umask_decorator "${FUNCNAME[0]}" "${@:-}" && return
+
+    debug 8 'Initializing NSS Wrapper'
+    assert test -n "${BUILD_GUID}"
+
+    export TMP_USER="${1:-bob}"
+    export TMP_GROUP="${2:-builders}"
+    tmp_passwd_file="$(mktemp -t "passwd.${$}.XXXXXXXXXX")" && add_on_exit "rm -f '${tmp_passwd_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_passwd_file}"
+    tmp_group_file="$(mktemp -t "group.${$}.XXXXXXXXXX")" && add_on_exit "rm -f '${tmp_group_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_group_file}"
+    tmp_hosts_file="$(mktemp -t "hosts.${$}.XXXXXXXXXX")" && add_on_exit "rm -f '${tmp_hosts_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_hosts_file}"
+
+    if [ -n "${3:-}" ] ; then
+        TMP_HOME_PATH="${3}"
+    else
+        TMP_HOME_PATH="$(mktemp -t -d "home.${TMP_USER}.XXXXXXXXXX")" && add_on_exit "rm -Rf '${TMP_HOME_PATH}'" && chown -R "${BUILD_GUID}" "${TMP_HOME_PATH}"
```

Comment:
Technically the `-d` should be before `-t`

Reply:
I'll swap those around for OSX compat.

Reply:
Shouldn't it be the case for Linux as well?